Content.Server/Blob/BlobCoreSystem.cs

```diff
@@ -0,0 +1,338 @@
+                    Dirty(blobbernautComponent);
+
+                    if (TryComp<MeleeWeaponComponent>(blobFactoryComponent.Blobbernaut, out var meleeWeaponComponent))
+                    {
+                        var blobbernautDamage = new DamageSpecifier();
+                        foreach (var keyValuePair in component.ChemDamageDict[component.CurrentChem].DamageDict)
+                        {
+                            blobbernautDamage.DamageDict.Add(keyValuePair.Key, keyValuePair.Value * 0.8f);
+                        }
+                        meleeWeaponComponent.Damage = blobbernautDamage;
+                    }
+
+                    ChangeBlobEntChem(blobFactoryComponent.Blobbernaut.Value, oldChem, newChem);
+                }
+
+                foreach (var compBlobPod in blobFactoryComponent.BlobPods)
+                {
+                    if (TryComp<SmokeOnTriggerComponent>(compBlobPod, out var smokeOnTriggerComponent))
+                    {
+                        smokeOnTriggerComponent.SmokeColor = component.ChemСolors[newChem];
+                    }
+                }
+            }
+
+            ChangeBlobEntChem(blobTile, oldChem, newChem);
+        }
+    }
+
+    private void OnDestruction(EntityUid uid, BlobCoreComponent component, DestructionEventArgs args)
+    {
+        if (component.Observer != null)
+        {
+            QueueDel(component.Observer.Value);
+        }
+
+        foreach (var blobTile in component.BlobTiles)
+        {
+            if (!TryComp<BlobTileComponent>(blobTile, out var blobTileComponent))
+                continue;
+            blobTileComponent.Core = null;
+
+            blobTileComponent.Color = Color.White;
+            Dirty(blobTileComponent);
+        }
+    }
+
+    private void ChangeBlobEntChem(EntityUid uid, BlobChemType oldChem, BlobChemType newChem)
+    {
+        var explosionResistance = EnsureComp<ExplosionResistanceComponent>(uid);
+        if (oldChem == BlobChemType.ExplosiveLattice)
+        {
+            _explosionSystem.SetExplosionResistance(uid, 0.3f, explosionResistance);
+        }
+        switch (newChem)
+        {
+            case BlobChemType.ExplosiveLattice:
+                _damageable.SetDamageModifierSetId(uid, "ExplosiveLatticeBlob");
+                _explosionSystem.SetExplosionResistance(uid, 0f, explosionResistance);
+                break;
+            case BlobChemType.ElectromagneticWeb:
+                _damageable.SetDamageModifierSetId(uid, "ElectromagneticWebBlob");
+                break;
+            case BlobChemType.ReactiveSpines:
+                _damageable.SetDamageModifierSetId(uid, "ReactiveSpinesBlob");
+                break;
+            default:
+                _damageable.SetDamageModifierSetId(uid, "BaseBlob");
+                break;
+        }
+    }
+
+    public bool TransformBlobTile(EntityUid? oldTileUid, EntityUid coreTileUid, string newBlobTileProto,
```

Comment:
Why is this blob core behavior. Much of this should be on tiles, blob core is barely related to most of this (copy pasted) code.